src/classes/COFF_Affiliation_TDTM.cls

```diff
@@ -168,14 +168,15 @@ public class COFF_Affiliation_TDTM extends TDTM_Runnable {
                 for (integer i=0; i<oldlist.size(); i++) {
                     Course_Offering__c oldOff = (Course_Offering__c)oldlist[i];
 
-                    if(triggerAction == TDTM_Runnable.Action.AfterDelete) {
+                    if(triggerAction == TDTM_Runnable.Action.BeforeDelete) {
 
                         //Course Offering has been deleted
                         //     --> delete related Affiliation, if the faculty field had a value
                         //     --> delete the Course Enrollment, if the faculty field had a value
                         if(oldOff.Faculty__c != null) {
                             UTIL_Debug.debug('****Course Offering with faculty has been deleted');
                             facultyCleanupIDs.add(oldOff.Faculty__c);
+                            offsWithRemovedFaculty.add(oldOff.Id);
```

Comment:
Clarification: ID field is not null here, but querying for these IDs won't return the expected records in after delete.